keel-lemur/src/main/kotlin/com/netflix/spinnaker/config/LemurConfiguration.kt

```diff
@@ -49,6 +54,27 @@ class LemurConfiguration {
       .build()
       .create(LemurService::class.java)
   }
+
+  @Bean
+  fun lemurCertificateByName(cacheFactory: CacheFactory, lemurService: LemurService) : suspend (String) -> LemurCertificateResponse {
+    val cacheName = "lemurCertificatesByName"
+    val cache = cacheFactory
+      .asyncLoadingCache<String, LemurCertificateResponse>(cacheName) { name ->
+        runCatching {
+          lemurService.certificateByName(name)
+        }
+          .getOrElse { ex ->
+            if (ex.isNotFound) {
+              null
+            } else {
+              throw CacheLoadingException("Error loading $cacheName cache", ex)
+            }
+          }
+      }
+    return {
+      cache.get(it).await()
+    }
+  }
```

Comment:
This creates a cache function wrapping the Lemur service call

Reply:
A closure! Nice...